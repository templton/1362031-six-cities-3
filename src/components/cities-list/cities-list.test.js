import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import CitiesList from "./cities-list";

const mockStore = configureStore([]);

describe(`Cities list render test`, () => {

  it(`Simple render`, () => {
    const city = {
      id: 17,
      name: `Амстердам`,
      cord: [52.38333, 4.9]
    };

    const cities = [
      {
        id: 17,
        name: `Амстердам`,
        cord: [52.38333, 4.9]
      },
      {
        id: 15,
        name: `Барнаул`,
        cord: [53.346785, 83.776856]
      },
      {
        id: 16,
        name: `Новосибирск`,
        cord: []
      }
    ];

    const store = mockStore({
      citiesList: cities,
      city
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <CitiesList />
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();

  });
});
