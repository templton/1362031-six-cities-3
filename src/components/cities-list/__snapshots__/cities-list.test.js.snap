// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <CitiesList/> Empty render 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  />
</section>
`;
